code/go/0chain.net/validatorcore/storage/writemarker/entity_test.go

```diff
@@ -0,0 +1,134 @@
+package writemarker_test
+
+import (
+	"fmt"
+	"testing"
+
+	"0chain.net/core/common"
+	"0chain.net/core/config"
+	"0chain.net/core/encryption"
+	"0chain.net/validatorcore/storage/writemarker"
+
+	"github.com/0chain/gosdk/core/zcncrypto"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestWriteMarker_GetHashData(t *testing.T) {
+	wm, wallet, err := setupEntityTest(t)
+	require.NoError(t, err)
+
+	want := fmt.Sprintf("%v:%v:%v:%v:%v:%v:%v", "alloc_root", "prev_alloc_root", "alloc_id", "blobber_id", wallet.ClientID, 1, common.Now())
+	got := wm.GetHashData()
+	if got != want {
+		t.Errorf("want %s, got %s", want, got)
+	}
```

Comment:
require.Equal(want, got) might be better.

Reply:
updated